### PR TITLE
Proposed fix for the issue #1 - https://github.com/serilog-web/classi…

### PR DESCRIPTION
**Pull request by konste**
Proposed fix for the issue #1 

.nuspec version should be incremented, but I don't know which mechanism Serilog uses for that. Currently $version$ token is used, but AssemblyInformationVersion is 1.0.0 and current version of the package is 2.02, so I don't know how to achieve 2.0.3

Konstantin

### CODE REVIEW

**Comment by IanYates** [comment]
:+1: - reading through it makes sense to me.  This is a really welcome change.  
I'd done something similar when trying to augment some of the built-in enrichers and ended up using [DebuggerNotMyCode] and [DebuggerStepThrough] although I suspect only the former is strictly needed to avoid the exceptions.

File: assets/CommonAssemblyInfo.cs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 using System.Reflection;
 
-[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyVersion("2.0.0.0")]
```
Comment (nblumhardt) [inline assets/CommonAssemblyInfo.cs]: This needs to stay fixed at `1.0.0.0` - we substitute it dynamically in the _AppVeyor_ build.

File: src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs [hunk 2]
```diff
@@ -128,7 +129,7 @@ static void LogRequest(object sender, EventArgs e)
         {
             if (!_isEnabled) return;
 
-            var request = HttpContext.Current.Request;
+            var request = HttpContextCurrent.Request;
             Logger.Write(_requestLoggingLevel, "HTTP {Method} for {RawUrl}", request.HttpMethod, request.RawUrl);
```
Comment (nblumhardt) [inline src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs:136]: After the change, this is going to lead to a `NullReferenceException` for `request.HttpMethod` etc., correct? Perhaps with the benefit of hindsight, going for the uglier (but more obviously null-returning) `TryGetCurrentRequest()` is a safer option?

File: src/SerilogWeb.Classic/Classic/Enrichers/HttpRequestClientHostIPEnricher.cs [hunk 3]
```diff
@@ -69,10 +69,10 @@ public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
             if (HttpContext.Current == null)
                 return;
 
-            if (HttpContext.Current.Request == null)
+            if (HttpContextCurrent.Request == null)
```
Comment (nblumhardt) [inline src/SerilogWeb.Classic/Classic/Enrichers/HttpRequestClientHostIPEnricher.cs:72]: In this instance, using:

```
HttpRequest request;
if (!HttpContextCurrent.TryGetRequest(out request))
    return;
```

Will have the added benefit that the follow-up calls can use the `request` variable and avoid calling the property repeatedly (not a significant performance hit, but still seems cleaner to make the null checks etc. only once).

File: src/SerilogWeb.Classic/SerilogWeb.Classic.nuspec [hunk 4]
```diff
@@ -2,7 +2,7 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
   <metadata>
     <id>SerilogWeb.Classic</id>
-    <version>$version$</version>
+    <version>2.0.3-private2</version>
```
Comment (nblumhardt) [inline src/SerilogWeb.Classic/SerilogWeb.Classic.nuspec]: Builds run through _AppVeyor_ so this is all done automatically. Needs a revert here as package versioning won't work for us with this change. To pack this spec with the `$version$` token in there, you can pass the `-Version` parameter to `nuget.exe pack` IIRC.

Comment (nblumhardt) [inline src/SerilogWeb.Classic/SerilogWeb.Classic.nuspec]: Actually - `Build.ps1` will do this for you - it's the kick-off point for the build.

**Comment by nblumhardt** [comment]
Thanks @konste! Added some comments in there. Cheers!

**Comment by konste** [comment]
I'm very new to Git and the concept of Pull Request got me confused. I have forked this repo, worked a little in my own branch then checked in, pushed to Github and created Pull Request. My understanding was that only changes I did before the Pull Request are included in it, so I continued modifying my own branch for my own purposes, like changing AssemblyVersion, etc. To my surprise it looks like all the changes (including some embarrassing ones) I did in my branch are still visible as part of the same first Pull Request! 

How can I disconnect my branch from Pull Request? Or what's the right workflow I should follow?

Konstantin

File: assets/CommonAssemblyInfo.cs (hunk 1 above)
Comment (konste) [inline assets/CommonAssemblyInfo.cs]: Change rolled back.

File: src/SerilogWeb.Classic/SerilogWeb.Classic.nuspec (hunk 4 above)
Comment (konste) [inline src/SerilogWeb.Classic/SerilogWeb.Classic.nuspec]: Change rolled back.

File: src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs (hunk 2 above)
Comment (konste) [inline src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs:136]: This is beyond the scope of the original problem and original fix, but I don't see a problem of fixing this on the way.

File: src/SerilogWeb.Classic/Classic/Enrichers/HttpRequestClientHostIPEnricher.cs (hunk 3 above)
Comment (konste) [inline src/SerilogWeb.Classic/Classic/Enrichers/HttpRequestClientHostIPEnricher.cs:72]: In this case it seems sufficient to remove the first check. I miss null coalescing operator from C# 6.0 so much!

**Comment by nblumhardt** [comment]
@konste what you need to do is:
1. Get the repo into a state you're happy with
2. Use git to "squash" the commits: https://www.google.com.au/webhp?q=git%20squash%20commits

Squashing will nuke any intermediate history, making the whole thing into a single rolled-up commit.

**Comment by konste** [comment]
Ok, I tried! Please see if it is good to pull in now.

**Comment by nblumhardt** [comment]
LGTM! :+1:

**Pull request merged**